test/extended/cmd.sh

```diff
@@ -71,75 +72,69 @@ popd
 rm -rf $tmp
 
 # ensure a local-only image gets a docker image(not imagestream) reference created.
-[ "$(oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' ' | grep -E "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+")" ]
+os::cmd::expect_success_and_text "oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' '" "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+"
```

Comment:
> The template to grab the right item out of a kapi.List and index into it will be pretty hairy... this may actually be less awful.

Try the jsonpath template.  If the predicates don't work right, that'd be good feedback to give upstream.